packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -0,0 +1,164 @@
+import * as t from '@babel/types';
+import kebabCase from '@compiled/ts-transform-css-in-js/dist/utils/kebab-case';
+import { addUnitIfNeeded } from '@compiled/ts-transform-css-in-js/dist/utils/css-property';
+import {
+  cssAfterInterpolation,
+  cssBeforeInterpolation,
+} from '@compiled/ts-transform-css-in-js/dist/utils/string-interpolations';
+import { hash } from '@compiled/ts-transform-css-in-js/dist/utils/hash';
+import generate from '@babel/generator';
+import { joinExpressions } from './ast-builders';
+import { State } from '../types';
+
+export interface CSSOutput {
+  css: string;
+  variables: {
+    name: string;
+    expression: t.Expression;
+  }[];
+}
+
+const getInterpolation = <TNode extends {}>(expression: TNode | undefined, state: State) => {
+  if (t.isIdentifier(expression) && state.declarations) {
+    const declaration = state.declarations[expression.name];
+    if (t.isVariableDeclaration(declaration) && declaration.kind === 'const') {
+      // We only want to pick out variable declarations that are constants
+      const potentialValue = declaration.declarations[0].init;
+      if (
+        t.isStringLiteral(potentialValue) ||
+        t.isNumericLiteral(potentialValue) ||
+        t.isObjectExpression(potentialValue)
+      ) {
+        return potentialValue;
+      }
+    }
+  }
+
+  return expression;
+};
+
+const normalizeContentValue = (value: string) => {
+  if (value.charAt(0) !== '"' && value.charAt(0) !== "'") {
+    return `"${value}"`;
+  }
+
+  return value;
+};
+
+const extractObjectExpression = (node: t.ObjectExpression, state: State): CSSOutput => {
+  let variables: CSSOutput['variables'] = [];
+  let css = '';
+
+  node.properties.forEach((prop) => {
+    if (t.isObjectProperty(prop)) {
+      // Don't use prop.value directly as it extracts constants from identifiers if needed.
+      const propValue = getInterpolation(prop.value, state);
+      const key = prop.key.name || prop.key.value;
+      let value = '';
+
+      if (t.isStringLiteral(propValue)) {
```

Comment:
lets add comments with example code for each block to help visualize whats happening